driver-async/src/main/com/mongodb/async/client/OperationIterable.java

```diff
@@ -117,6 +117,12 @@ public void onResult(final List<T> results, final Throwable t) {
         return new MappingIterable<T, U>(this, mapper);
     }
 
+    @Override
+    public OperationIterable<T> batchSize(final int batchSize) {
```

Comment:
This looks like a problem.  For example:

     collection.find(...).map(...).batchSize(6).forEach(...)

will silently drop the batch size.